File: hh_aggregate_service/src/main/java/by/icemens/hh_aggregate_service/model/Vacancy.java
package by.icemens.hh_aggregate_service.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Модель данных вакансии.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Vacancy {
    
    /**
     * Заголовок вакансии.
     */
    private String title;
    
    /**
     * URL вакансии.
     */
    @JsonProperty("url")
    private String url;
    
    /**
     * Работодатель.
     */
    private String employer;
    
    /**
     * Описание вакансии.
     */
    private String description;
    
    /**
     * Зарплатная вилка (минимум).
     */
    @JsonProperty("salary_from")
    private Integer salaryFrom;
    
    /**
     * Зарплатная вилка (максимум).
     */
    @JsonProperty("salary_to")
    private Integer salaryTo;
    
    /**
     * Валюта зарплаты.
     */
    private String currency;
    
    /**
     * Регион.
     */
    private String region;
}

File: hh_autoapply_service/internal/model/vacancy.go
package model

type Vacancy struct {
	ID          int64  `json:"id" db:"id"`
	Title       string `json:"title" db:"title"`
	Employer    string `json:"employer" db:"employer"`
	URL         string `json:"url" db:"url"`
	Description string `json:"description,omitempty" db:"description"`
	SalaryFrom  *int64 `json:"salaryFrom,omitempty" db:"salary_from"`
	SalaryTo    *int64 `json:"salaryTo,omitempty" db:"salary_to"`
	Currency    string `json:"currency,omitempty" db:"currency"`
	Region      string `json:"region,omitempty" db:"region"`
	UserID      int64  `json:"user_id" db:"user_id"`
}
